itest/assertions.go

```diff
@@ -142,6 +142,72 @@ func AssetScriptKeyIsBurnCheck(isBurn bool) AssetCheck {
 	}
 }
 
+func AssetScriptKeyCheck(scriptKey *taprpc.ScriptKey) AssetCheck {
+	return func(a *taprpc.Asset) error {
+		if scriptKey != nil {
```

Comment:
nit: invert condition and return early to continue un-indented below? Same for the other new checks below.